From e7bfb246447a1c2a6249c20f0a006ad1578def01 Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Fri, 8 Dec 2023 11:47:43 +0000
Subject: [PATCH] Don't pass unknown LLVM options

Because we compile and link with an older LLVM version than upstream, we
need to drop the following LLVM options that aren't available yet:
* disable-auto-upgrade-debug-info
* split-threshold-for-reg-with-hint

Trying to pass them anyways will lead to an "Unknown command line
argument" error.

Upstream-Status: Inappropriate [specific to older versions of LLVM]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 build/config/compiler/BUILD.gn | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 80bc79e..715bffc 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -616,24 +616,6 @@ config("compiler") {
       }
     }

-    # TODO(crbug.com/1488374): This causes binary size growth and potentially
-    # other problems.
-    # TODO(crbug.com/1491036): This isn't supported by Cronet's mainline llvm version.
-    if (default_toolchain != "//build/toolchain/cros:target" &&
-        !llvm_android_mainline) {
-      cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
-      ]
-      if (use_thin_lto && is_a_target_toolchain) {
-        if (is_win) {
-          ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
-        } else {
-          ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
-        }
-      }
-    }
-
     # TODO(crbug.com/1235145): Investigate why/if this should be needed.
     if (is_win) {
       cflags += [ "/clang:-ffp-contract=off" ]
@@ -800,13 +782,6 @@ config("compiler") {
       if (is_apple) {
         ldflags += [ "-Wcrl,object_path_lto" ]
       }
-      if (!is_chromeos) {
-        # TODO(https://crbug.com/972449): turn on for ChromeOS when that
-        # toolchain has this flag.
-        # We only use one version of LLVM within a build so there's no need to
-        # upgrade debug info, which can be expensive since it runs the verifier.
-        ldflags += [ "-Wl,-mllvm,-disable-auto-upgrade-debug-info" ]
-      }
     }

     # TODO(https://crbug.com/1211155): investigate why this isn't effective on
